record_archiver/models/record_lifespan.py

```diff
@@ -0,0 +1,114 @@
+
+# Copyright 2015-2016 Yannick Vaucher (Camptocamp SA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+from openerp import api, exceptions, fields, models
+from openerp.tools.translate import _
+
+_logger = logging.getLogger(__name__)
+
+
+class RecordLifespan(models.Model):
+    """Configure records lifespans per model.
+
+    After the lifespan is expired (compared to the `write_date` of the
+    records), the records are deactivated.
+    """
+    _name = 'record.lifespan'
+    _order = 'model'
+
+    model_id = fields.Many2one(
+        'ir.model',
+        string='Model',
+        required=True,
+        domain=[('has_an_active_field', '=', True)],
+    )
+    model = fields.Char(
+        related='model_id.model',
+        string='Model Name',
+        store=True,
+    )
+    months = fields.Integer(
+        required=True,
+        help="Number of month after which the records will be set to inactive "
+             "based on their write date"
+    )
+
+    _sql_constraints = [
+        ('months_gt_0', 'check (months > 0)',
+         "Months must be a value greater than 0"),
```

Comment:
translate missing

Reply:
months is editable via UI right? If the user enters a bad value you gonna bump that error on her/his face, right?
If these assumptions are true: pls, translate.